src/rf/apps/workers/process.py

```diff
@@ -58,48 +63,84 @@ def start_polling(self):
                         delete_message = self.check_timeout(record['data'])
                     elif job_type in JOB_VALIDATE:
                         if record['eventSource'] == 'aws:s3':
-                            delete_message = self.validate_image(record)
+                            delete_message = self.__validate_image(record)
 
                 if delete_message:
                     self.q.delete_message(message)
 
-    def validate_image(self, data):
+    def __validate_image(self, data):
         """
         Use Gdal to verify an image is properly formatted and can be further
         processed.
         data -- attribute data from SQS.
         """
-        # Pass image to Gdal to verify.
-        # If it succeeds continue else return False.
-
-        key = data['s3']['object']['key']
-        payload_uuid = self.extract_uuid_from_aws_key(key)
-
-        url = self.make_s3_url(data['s3']['bucket']['name'], key)
 
-        # TODO - Right now we don't save data to the DB with upload. So hard
-        # code something for testing.
-        try:
-            image = LayerImage.objects.get(s3_uuid=payload_uuid)
+        def ensure_band_count(key_string, range=None):
+            """
+            Gets the first `range` bytes from the s3 resource and uses it to
+            determine the number of bands in the image.
+            """
+            connection = S3Connection()
+            bucket = connection.get_bucket(settings.AWS_BUCKET_NAME)
+            s3_key = Key(bucket)
+            s3_key.key = key_string
+            random_filename = '/tmp/' + str(uuid.uuid4())
```

Comment:
I think it might be good to setup a variable in the Django settings file, or somewhere else that all worker subtasks can see, with a scratch space directory. I think that directory should be `/media/ephemeral0` because EC2 instances with instance storage support automatically mount an `ext3` file system there that goes away on reboots. It is also local to the underlying virtual machine host vs. `/tmp` which would be network attached (via EBS).

For Vagrant, I think we need to add a conditional task scoped by environment that ensures `/media/ephemeral0` exists and is owned by `rf`. For EC2, I some entries need to be added to the worker `cloud-config` (housed within the CloudFormation):

```yams
runcmd:
 - [ chown, rf, "/media/ephemeral0" ]
 - [ chgrp, rf, "/media/ephemeral0" ]
```